dht.go

```diff
@@ -437,3 +507,49 @@ func (dht *IpfsDHT) newContextWithLocalTags(ctx context.Context, extraTags ...ta
 	) // ignoring error as it is unrelated to the actual function of this code.
 	return ctx
 }
+
+func (dht *IpfsDHT) handleProtocolChanges(ctx context.Context) {
+	// register for event bus protocol ID changes
+	ch := make(chan event.EvtPeerProtocolsUpdated, 8)
+	cancel, err := dht.host.EventBus().Subscribe(ch)
+	if err != nil {
+		panic(err)
+	}
+	defer cancel()
+
+	pmap := make(map[protocol.ID]bool)
+	for _, p := range dht.protocols {
+		pmap[p] = true
+	}
+
+	for {
+		select {
+		case e, ok := <-ch:
+			if !ok {
+				return
+			}
+			var drop, add bool
+			for _, p := range e.Added {
+				if pmap[p] {
+					add = true
+				}
+			}
+			for _, p := range e.Removed {
+				if pmap[p] {
+					drop = true
+				}
+			}
+
+			if add && drop {
+				// TODO: discuss how to handle this case
+				logger.Warning("peer adding and dropping dht protocols? odd")
```

Comment:
Still need to figure out what to do here, @Kubuxu  and @raulk any thoughts?

Reply:
Proposal: "Peer is bad, we don't want them in our routing table"

Reply:
We can either noop (since both operations cancel each other out), or discard the peer altogether.